port_ocean/clients/port/mixins/integrations.py

```diff
@@ -50,13 +51,36 @@ async def get_log_attributes(self) -> LogAttributes:
             self._log_attributes = response["logAttributes"]
         return self._log_attributes
 
+    async def _poll_integration_until_default_provisioning_is_complete(
+        self,
+        interval=15,
+    ) -> Dict[str, Any]:
+        response = await self._get_current_integration()
+        response_json = response.json()
+        config = response_json.get("integration", {}).get("config", {})
+        if config != {}:
+            return response_json
+
+        logger.info(
+            f"Still waiting for integration config to be ready, waiting {interval} seconds"
+        )
+        await asyncio.sleep(interval)
+
+        # TODO: Ensure that get_integration isn't cached
+        result = await self._poll_integration_until_default_provisioning_is_complete()
+
+        return result
+
     async def create_integration(
         self,
         _type: str,
         changelog_destination: dict[str, Any],
         port_app_config: Optional["PortAppConfig"] = None,
+        use_provisioned_defaults: Optional[bool] = False,
     ) -> dict:
         logger.info(f"Creating integration with id: {self.integration_identifier}")
+        if use_provisioned_defaults:
```

Comment:
this is redundant if, you already do the if later in the code and can put the log there

Reply:
Fixed!